File: leet_easy_c++/leet_066/leet_066b.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int len = digits.size(), cur = len-1;
        if(len == 0) return (vector<int>){1};
        digits[cur]++;
        for(;cur >= 0 && digits[cur]==10; cur--){
            digits[cur]=0;
            if(cur>0) digits[cur-1]++;
            else{
                digits.emplace_back(0);
                digits[0]=1;
            }
        }
        return digits;
    }
};

File: leet_easy_python/leet_066/leet_066b.py
class Solution(object):
    def plusOne(self, digits):
        sz_d = len(digits)
        j = sz_d-1
        while j>=0 and digits[j]==9: j-=1
        if j<0:
            digits = [0]*(sz_d+1)
            digits[0] = 1
        else:
            digits[j]+=1
            for x in range(j+1, sz_d): digits[x]=0 
        return digits
        
        """
        :type digits: List[int]
        :rtype: List[int]
        """